mmaction/datasets/ssn_dataset.py

```diff
@@ -0,0 +1,660 @@
+        if not mmcv.is_tuple_of(self.aug_ratio, (int, float)):
+            raise TypeError(f'aug_ratio should be int, float'
+                            f'or tuple of int and float, '
+                            f'but got {type(aug_ratio)}')
+        assert len(self.aug_ratio) == 2
+
+        total_ratio = self.sampler.fg_ratio + self.sampler.bg_ratio + \
+            self.sampler.incomplete_ratio
+        self.fg_per_video = int(self.sampler.num_per_video *
+                                (self.sampler.fg_ratio / total_ratio))
+        self.bg_per_video = int(self.sampler.num_per_video *
+                                (self.sampler.bg_ratio / total_ratio))
+        self.incomplete_per_video = self.sampler.num_per_video -\
+            self.fg_per_video - self.bg_per_video
+
+        self.test_interval = self.test_cfg.ssn.sampler.test_interval
+        # number of consecutive frames
+        self.clip_len = clip_len
+        # number of steps (sparse sampling for efficiency of io)
+        self.frame_interval = frame_interval
+
+        # test mode or not
+        self.filter_gt = filter_gt
+        self.test_mode = test_mode
+
+        if self.verbose:
+            print("""
+            SSNDataset: proposal file {prop_file} parsed.
+
+            There are {pnum} usable proposals from {vnum} videos.
+            {fnum} foreground proposals
+            {inum} incomplete proposals
+            {bnum} background proposals
+
+            Sample config:
+            FG/BG/INC: {fr}/{br}/{ir}
+            Video Centric: {vc}
+
+            Regression Stats:
+            Location: mean {stats[0][0]:.05f} std {stats[1][0]:.05f}
+            Duration: mean {stats[0][1]:.05f} std {stats[1][1]:.05f}
+            """.format(
+                prop_file=self.proposal_file,
+                pnum=len(self.fg_pool) + len(self.bg_pool) +
+                len(self.incomp_pool),
+                fnum=len(self.fg_pool),
+                inum=len(self.incomp_pool),
+                bnum=len(self.bg_pool),
+                fr=self.fg_per_video,
+                br=self.bg_per_video,
+                ir=self.incomplete_per_video,
+                vnum=len(self.video_infos),
+                vc=self.video_centric,
+                stats=self.reg_stats))
+        else:
+            print("""
+            SSNDataset: proposal file {prop_file} parsed.
+            """.format(prop_file=self.proposal_file))
+
+    def load_annotations(self):
+        """Load annotation file to get video information."""
+        video_infos = []
+        if 'normalized_' in self.ann_file:
+            self.proposal_file = self.ann_file.replace('normalized_', '')
+            if not osp.exists(self.proposal_file):
+                print('{} does not exist. Converting from {}'.format(
+                    self.proposal_file, self.ann_file))
+                frame_dict = parse_directory(
+                    self.data_prefix, key_func=lambda x: x.split('/')[-1])
```

Comment:
It seems `parse_directory` is not necessary, since we can use frame annotation file to directly get the frame number.